contracts/governance/utils/VotesConfidential.sol

```diff
@@ -0,0 +1,202 @@
+     *
+     * Requirements:
+     *
+     * - `timepoint` must be in the past. If operating using block numbers, the block must be already mined.
+     */
+    function getPastTotalSupply(uint256 timepoint) public view virtual returns (euint64) {
+        return _totalCheckpoints.upperLookupRecent(_validateTimepoint(timepoint));
+    }
+
+    /**
+     * @dev Returns the current total supply of votes as an encrypted uint64 (euint64). Must be implemented
+     * by the derived contract.
+     */
+    function totalSupply() public view virtual returns (euint64);
+
+    /// @dev Returns the delegate that `account` has chosen.
+    function delegates(address account) public view virtual returns (address) {
+        return _delegatee[account];
+    }
+
+    /// @dev Delegates votes from the sender to `delegatee`.
+    function delegate(address delegatee) public virtual {
+        _delegate(msg.sender, delegatee);
+    }
+
+    /// @dev Delegates votes from an EOA to `delegatee` via an ECDSA signature.
+    function delegateBySig(
+        address delegatee,
+        uint256 nonce,
+        uint256 expiry,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public virtual {
+        if (block.timestamp > expiry) {
+            revert VotesExpiredSignature(expiry);
+        }
+
+        address signer = ECDSA.recover(
+            _hashTypedDataV4(keccak256(abi.encode(DELEGATION_TYPEHASH, delegatee, nonce, expiry))),
+            v,
+            r,
+            s
+        );
+
+        _useCheckedNonce(signer, nonce);
+        _delegate(signer, delegatee);
+    }
+
+    /**
+     * @dev Delegate all of `account`'s voting units to `delegatee`.
+     *
+     * Emits events {IVotes-DelegateChanged} and {IVotes-DelegateVotesChanged}.
+     */
+    function _delegate(address account, address delegatee) internal virtual {
+        address oldDelegate = delegates(account);
+        _delegatee[account] = delegatee;
+
+        emit DelegateChanged(account, oldDelegate, delegatee);
+        _moveDelegateVotes(oldDelegate, delegatee, _getVotingUnits(account));
+    }
+
+    /**
+     * @dev Transfers, mints, or burns voting units. To register a mint, `from` should be zero. To register a burn, `to`
+     * should be zero. Total supply of voting units will be adjusted with mints and burns.
+     *
+     * WARNING: Must be called after {totalSupply} is updated.
+     */
+    function _transferVotingUnits(address from, address to, euint64 amount) internal virtual {
+        if (from == address(0) || to == address(0)) {
+            _push(_totalCheckpoints, totalSupply());
+        }
+        _moveDelegateVotes(delegates(from), delegates(to), amount);
+    }
+
+    /**
+     * @dev Moves delegated votes from one delegate to another.
+     */
+    function _moveDelegateVotes(address from, address to, euint64 amount) internal virtual {
+        CheckpointsConfidential.TraceEuint64 storage store;
+        if (from != to && euint64.unwrap(amount) != 0) {
```

Comment:
`        if (from != to && amount.isInitialized()) {` (& everywhere).
I can create a PR @arr00 if you agree with that (?)

Reply:
Feel free to create a PR. Seems like `isInitialized` does the same thing but is a nicer semantic.